bin/full-node/src/run/network_service.rs

```diff
@@ -665,6 +665,10 @@ pub enum InitError {
     #[display(fmt = "I/O error when creating listener for {}: {}", _0, _1)]
     ListenerIo(Multiaddr, io::Error),
     /// A listening address passed through the configuration isn't valid.
+    #[display(
```

Comment:
[`thiserror`](https://docs.rs/thiserror/latest/thiserror/) only applies to errors in opposite to `derive_more` which is more general and provides more functionality, but they are more or less the same thing, right?

Reply:
Yes. `thiserror` doesn't support no-std platforms though.